src/commands/staff/mute.ts

```diff
@@ -10,23 +9,43 @@ export default class extends Command {
 	usage = '<user>';
 	runInDM = false;
 
-	permissions(msg: Message): boolean {
-		return staffPerms(msg);
-	}
+	tempPermissions: ApplicationCommandPermissionData[] = [STAFF_PERMS, ADMIN_PERMS];
+
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'user',
+			description: 'The user to mute',
+			type: 'USER',
+			required: true
+		}
+	]
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		const user = interaction.options.getUser('user');
+		const member = await interaction.guild.members.fetch(user.id);
+
+		if (!member) {
+			interaction.reply({
+				content: `Something went wrong. Please contact ${MAINTAINERS} for help. You can give the user the <@&${ROLES.MUTED}> role manually as a quick fix.`,
+				ephemeral: true
+			});
+			throw new Error('Could not find member based on passed in user');
+		}
 
-	async run(msg: Message, [member]: [GuildMember]): Promise<Message> {
 		if (member.roles.cache.has(ROLES.MUTED)) {
-			const reason = `${member.user.username} was un-muted by ${msg.author.tag} (${msg.author.id})`;
-			member.roles.remove(ROLES.MUTED, reason);
-			return msg.channel.send(`${member.user.username} has been un-muted.`);
+			const reason = `${member.user.username} was un-muted by ${interaction.user.tag} (${interaction.user.id})`;
+			await member.roles.remove(ROLES.MUTED, reason);
+			return interaction.reply({ content: `${member.user.username} has been un-muted.`, ephemeral: true });
 		}
-		const reason = `${member.user.username} was muted by ${msg.author.tag} (${msg.author.id})`;
-		member.roles.add(ROLES.MUTED, reason);
-		return msg.channel.send(`${member.user.username} has been muted.`);
+		const reason = `${member.user.username} was muted by ${interaction.user.tag} (${interaction.user.id})`;
+		await member.roles.add(ROLES.MUTED, reason);
```

Comment:
You might want to make use of Discord's new [timeout feature](https://discord.js.org/#/docs/discord.js/stable/class/GuildMember?scrollTo=timeout).

Reply:
There is an issue with this - if a user is timed out, they cannot use _any_ channel. We discussed the mute policy a few days ago and have decided to stick with a muted role so, if a user breaks the rules in a non-course channel, they are still able to interact in academic channels but not in non-course channels like #general. Punishments for any rule breakages in academic channels would be down to professor discretion.